modules/packages/EpochManager.chpl

```diff
@@ -0,0 +1,1147 @@
+/*
+ * Copyright 2004-2019 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+
+To use the :class:`EpochManager`, first create an instance.
+
+.. code-block:: chapel
+
+ var manager = new unmanaged EpochManager();
+
+
+Registering a Task
+------------------
+A task must be registered with the manager in order to use the manager.
+Registration returns a token.
```

Comment:
Registration returns a token. -> Registration returns a token that is used when a task inters or exits a critical section.